modules/apm/src/main/java/METERING.md

```diff
@@ -0,0 +1,171 @@
+We use elastic's apm-java-agent as an implementation of the API we expose.
+the implementation can be found in `:modules:apm`
+The apm-java-agent is responsible for buffering metrics and upon metrics_interval
+send them over to apm server.
+Metrics_interval is configured via a `tracing.apm.agent.metrics_interval` setting
+The agent also collects a number of JVM metrics.
+see https://www.elastic.co/guide/en/apm/agent/java/current/metrics.html#metrics-jvm
+
+
+## How to choose an instrument
+
+We support various instruments and might be adding more as we go.
+The choice of the right instrument is not always easy as often differences are subtle.
+The simplified algorithm could be as follows:
+
+1. You want to measure something (absolute value)
+    1. values are non-additive
+        1.  use a gauge
+        2.  Example: a cpu temperature
+    2. values are additive
+        1. use asynchronous counter
+        2. Example: total number of requests
+2. You want to count something
+    1. values are monotonously increasing
+        1. use a counter
+        2. Example: Recording a failed authentication count
+    2. values can be decreased
+        1. use UpDownCounter
+        2. Example: Number of orders in a queue
+3. You want to record a statistics
+    1. use a histogram
+        1. Example: A statistics about how long it took to access a value from cache
+
+refer to https://opentelemetry.io/docs/specs/otel/metrics/supplementary-guidelines/#instrument-selection
+for more details
+
+## How to name an instrument
+See the naming guidelines for metrics:
+https://docs.google.com/document/d/1jKxuaZi7QAMIRD_Eq3nonkYswVlQXW5TWllJEicoOtM/edit#heading=h.jxn90hx2ayic
+
+### Restarts and overflows
+if the instrument is correctly chosen, the apm server will be able to determine if the metrics
+were restarted (i.e. node was restarted) or there was a counter overflow
+(the metric in ES might use an int internally, but apm backend might have a long )
+
+## How to use an instrument
+There are 2 types of usages of an instrument depending on a type.
+- For synchronous instrument (counter/UpDownCounter) we need to register an instrument with
+  `MeterRegistry` and use the returned value to increment a value of that instrument
+```java
+  MeterRegistry registry;
+  LongCounter longCounter = registry.registerLongCounter("es.test.requests.count", "a test counter", "count");
+  longCounter.increment();
+  longCounter.incrementBy(1, Map.of("name", "Alice"));
+  longCounter.incrementBy(1, Map.of("name", "Bob"));
+```
+
+- For asynchronous instrument (gauge/AsynchronousCounter) we register an instrument
+  and have to provide a callback that will report the absolute measured value.
+  This callback has to be provided upon registration and cannot be changed.
+```java
+MeterRegistry registry;
+long someValue = 1;
+registry.registerLongGauge("es.test.cpu.temperature", "a test gauge", "celcius",
+() -> new LongWithAttributes(someValue, Map.of("cpuNumber", 1)));
```

Comment:
Indent.

Description: "the current CPU temperature as measured by psensor"
Unit: "degrees Celsius"

Reply:
it is within the ```java so if I add extra tab it would look odd in the generated view (in my opinion)
updated the desc and unit